tfe/resource_tfe_registry_module.go

```diff
@@ -190,6 +198,41 @@ func resourceTFERegistryModuleCreate(d *schema.ResourceData, meta interface{}) e
 	return resourceTFERegistryModuleRead(d, meta)
 }
 
+func resourceTFERegistryModuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	options := tfe.RegistryModuleUpdateOptions{
+		NoCode: tfe.Bool(d.Get("no_code").(bool)),
+	}
+	var registryModule *tfe.RegistryModule
+	var err error
+
+	rmID := tfe.RegistryModuleID{
+		Organization: d.Get("organization").(string),
+		Name:         d.Get("name").(string),
+		Provider:     d.Get("module_provider").(string),
+		Namespace:    d.Get("namespace").(string),
+		RegistryName: tfe.RegistryName(d.Get("registry_name").(string)),
+	}
+
+	err = resource.Retry(time.Duration(5)*time.Minute, func() *resource.RetryError {
+		registryModule, err = tfeClient.RegistryModules.Update(ctx, rmID, options)
+		if err != nil {
+			return resource.RetryableError(err)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error while waiting for module %s/%s to be ingested: %w", registryModule.Organization.Name, registryModule.Name, err)
+	}
+
+	d.SetId(registryModule.ID)
+	d.Set("no_code", registryModule.NoCode)
+
+	return nil
```

Comment:
Typically an `Update()` will return with a call to `Read()`. While we aren't explicit about this, `Read()` should be the only method that should be used to write attributes to state (with the exception of the resource ID). 

Line 231 should be added to `Read()` instead.

Reply:
I did realize that `Create()` writes certain attributes to state, but only because `Read()` depends on these values. Given that `Update()` will only ever be called if the `no_code` attribute is used as an argument (since all other attributes are `ForceNew: true`), we can get away with using `Read()` to write the `no_code` attribute.

Reply:
And final question, do we have to wait for the module to be ingested if it's not being created?

Reply:
> And final question, do we have to wait for the module to be ingested if it's not being created?

Evils of copy-pasting 😅 everything has been addressed now 👍🏼